filebeat/prospector/prospector.go

```diff
@@ -147,6 +137,26 @@ func (p *Prospector) Run() {
 	}
 }
 
+// updateState updates the prospector state and forwards the event to the spooler
+// All state updates done by the prospector itself are synchronous to make sure not states are overwritten
+func (p *Prospector) updateState(event *input.Event) {
+	p.stateMutex.Lock()
+	defer p.stateMutex.Unlock()
+
+	// Add ttl if cleanOlder is enabled
+	if p.config.CleanInactive > 0 {
+		event.State.TTL = p.config.CleanInactive
+	}
+
+	ok := p.outlet.OnEvent(event)
```

Comment:
we want to filter out 'empty' events here or are empty events required for some state update somehwere else?